### ISSUE TITLE
Resurrect Zen mode (only show selected window)

### ISSUE THREAD

**Issue by olejorgenb** [paperwm/PaperWM#150]
![Screenshot from 2019-07-06 12-57-00](https://user-images.githubusercontent.com/72201/60755272-aa9d1e80-9fed-11e9-97ec-759add1052f1.png)

Center the focused window and push other windows to the side. Maybe hide topbar too?

As discussed earlier a good way to activate is double centering the window. That way we don't need a new binding. Recenter a "zen window" or navigating would exit the mode.

**Comment by olejorgenb** [comment]
A very MVP implementation: https://github.com/paperwm/PaperWM/tree/mvp-zen-mode

**Comment by eliorc** [comment]
Thanks for rightly referencing this from #367, I love this!

But I do think the functionality has to be able to extend to more than one window as much tasks are multiple windows native. Developing + Stackoverflow, writing FE code and watching the render in live in a browser etc.

**Comment by markk** [comment]
Would you not just use workspaces in that case?

**Comment by hedning** [comment]
Another  workspace could be an ad-hoc fix, but one of the goals we try to achieve is having a meaningful separation between workspaces, ie. a workspace should be able to contain all work related to that topic.

So I agree selecting several windows for a zen mode could be nice.

Now personally there's others things I want to do before a zen-mode is properly tackled, and I've sketched down  somewhat of a road map of the things I think we should really add/get done and should probably just post and pin the road map as I see it (eg. navigating the minimap with the mouse, like the alt-tab preview, is something I'd really like to get done, also #225 is something I use and really makes mouse interactions alot better).

**Comment by eliorc** [comment]
@markk as I see it workspaces are not temporary, but rather permanent - so while working, I might keep my WhatsApp web browser and my personal email on one workspace, while the work related stuff on another. As I see it Zen mode is more of a temporary configuration for what I am doing right now inside a workspace, like replying to an email, fixing a bug etc.

**Comment by eliorc** [comment]
![image](https://user-images.githubusercontent.com/17727283/117925598-2bbb5000-b300-11eb-9351-191bf0eabc3c.png)

**Comment by eliorc** [comment]
I think it's worth reminding that this feature would be an awesome addition to PaperWM, when I code and want 100% focus today I move my IDE to a separate workspace. The problem with that is that when I'm reading some documentation, or consulting SO/GPT I find myself flicking back and forward which gets quite complex to get right when you have multiple windows open.
I think the ability to "Zen" mode into a specific window, un-Zen it for a second to see the whole context and then re-Zen it would be a good productivity booster

**Comment by jtaala** [comment]
Right, it's about time we implemented this.  @eliorc has been waiting long enough!

I've started work on this.  Will report back when I've got something to test.

**Comment by eliorc** [comment]
psst psst 👀

**Comment by eliorc** [comment]
For other people visiting here, I use [this extension](https://extensions.gnome.org/extension/6313/dim-background-windows/) to get the desired effect

**Comment by smichel17** [comment]
Search keyword: focus mode
